Recommendations

ReViZe96 opened the pull request.

ReViZe96 committed:
0.3.1 explore-with-me:
1. create collector (need testing)
2. add proto and avro schemas for collector's entities and service
3. add kafka to docker-compose.yml
4. add pom's and empty configs for aggregator and analyzer

ReViZe96 committed:
0.3.2 explore-with-me:
1. add aggregator structure

ReViZe96 committed:
0.3.3 explore-with-me:
1. add analyzer service (need testing).
2. remove stats-server and stats-dto modules.
3. refactoring of core modules: set views on rating
4. add new methods to core services

ReViZe96 committed:
0.3.4 explore-with-me:
1. add aggregator service (need testing).

ReViZe96 committed:
0.3.4 explore-with-me:
1. server.port 0 for collector

ReViZe96 committed:
0.3.4 explore-with-me:
1. server.port 0 for analyzer

ReViZe96 committed:
0.3.4 explore-with-me:
1. key with void type for collector

ReViZe96 committed:
0.3.4 explore-with-me:

ReViZe96 committed:
0.3.4 explore-with-me:
1. disable linger.ms.config in collector kafka producer
2. set package for recommendations_controller.proto

ReViZe96 committed:
0.3.4 explore-with-me:
1. aggregator refactoring

ReViZe96 committed:
0.3.4 explore-with-me:
1. EnableFeignClient annotations for core modules's main classes
2. remove spring-retry dependency from core modules pom's

ReViZe96 committed:
0.3.4 explore-with-me:
1. EnableFeignClient annotations for category-service main classes
2. remove spring-retry dependency from category-servic pom's

ReViZe96 committed:
0.3.4 explore-with-me:
1. edit request-service

ReViZe96 committed:
0.3.4 explore-with-me:

ReViZe96 committed:
0.3.4 explore-with-me:

ReViZe96 committed:
0.3.4 explore-with-me:
manageOffsets() for aggregator

andrew-larkin commented on stats/aggregator/src/main/java/ru/practicum/service/AggregatorService.java:
```diff
@@ -0,0 +1,122 @@
+        Double score = 0d;
+        switch (actionType) {
+            case VIEW:
+                score = 0.4;
```
хорошо было бы вынести эти значения в application.yaml, а потом доставать их в конфиге, например, через @ConfigurationProperties

andrew-larkin commented on stats/analyzer/src/main/java/ru/practicum/consumer/AnalyzerTopics.java:
```diff
@@ -0,0 +1,8 @@
+
+public class AnalyzerTopics {
+
+    public final static String USER_TOPIC = "stats.user-actions.v1";
```
тоже лучше вынести в application.yaml

andrew-larkin commented on core/event-service/src/main/java/ru/practicum/event/controller/PublicEventController.java:
```diff
@@ -88,13 +82,30 @@ public List<EventShortDto> getEvents(@RequestParam(required = false) @Size(min =
+    }
+
     @GetMapping("/byIds")
     List<EventDto> getEvents(@RequestParam(required = false) List<Long> ids) {
```
модификаторы доступа

andrew-larkin reviewed (changes requested):
Привет! Хорошая работа! в нескольких местах вынести текстовые значения настроек в application.yaml, еще пара правок и приложение будет полностью готово.

ReViZe96 committed:
0.3.5 explore-with-me:
after review fix:
1. add java-doc for all controllers
2. score (user action's weight) value's configuring move to application.yaml
3. topic's configuring move to application.yaml
4. kafka configuration class for collector module

ReViZe96 committed:
0.3.5 explore-with-me:
unmute services in docker-compose.yml

ReViZe96 committed:
0.3.5 explore-with-me:

ReViZe96 committed:
0.3.5 explore-with-me:

ReViZe96 committed:
0.3.5 explore-with-me:
modify aggregator application.yaml (aggregator.kafka.score to aggregator.score)

ReViZe96 committed:
0.3.5 explore-with-me:

andrew-larkin reviewed (approved):
Привет! Отлично, теперь приложение полностью готово! Поздравляю с окончанием курса и желаю как можно скорее найти свою первую работу программистом!
